actors/builtin/multisig/multisig_actor.go

```diff
@@ -505,7 +514,7 @@ func getTransaction(rt runtime.Runtime, ptx *adt.Map, txnID TxnID, proposalHash
 	if checkHash {
 		calculatedHash, err := ComputeProposalHash(&txn, rt.HashBlake2b)
 		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to compute proposal hash for %v", txnID)
-		if proposalHash != nil && !bytes.Equal(proposalHash, calculatedHash[:]) {
+		if len(proposalHash) > 0 && !bytes.Equal(proposalHash, calculatedHash[:]) {
```

Comment:
I think this is making the method robust against ambiguous decoding of proposal hash into either `nil` or `[]byte{}` 👍 .  Out of curiosity did you actually run into this case from serialized params?  I would have expected cbor gen to have 1. the same encoding of both values 2. a consistent decoding of the serialization of both possible values.  In both cases picking one

Reply:
Yes this is true. I did not run into it directly, but traced through here while testing. The cbor-gen does look to produce a `nil`. Unit tests will have to consistently use nil too.

I'll revert for now.